spring-cloud-azure-appconfiguration-config/src/main/java/com/microsoft/azure/spring/cloud/config/AzureConfigPropertySource.java

```diff
@@ -43,12 +59,33 @@ public Object getProperty(String name) {
 
     public void initProperties() {
         // * for wildcard match
+        // Reading in Configurations
         QueryOptions queryOptions = new QueryOptions().withKeyNames(context + "*").withLabels(label);
         List<KeyValueItem> items = source.getKeys(storeName, queryOptions);
-
         for (KeyValueItem item : items) {
             String key = item.getKey().trim().substring(context.length()).replace('/', '.');
             properties.put(key, item.getValue());
         }
+
+        // Reading In Features
+        queryOptions = new QueryOptions().withKeyNames("*appconfig*").withLabels(label);
+        items = source.getKeys(storeName, queryOptions);
+        FeatureSet featureSet = new FeatureSet();
+        for (KeyValueItem item : items) {
+            FeatureManagementItem featureItem;
+            try {
+                featureItem = mapper.readValue(item.getValue(), FeatureManagementItem.class);
+                Feature feature = new Feature();
+                feature.setEnabled(featureItem.getEnabled());
+                feature.setId(featureItem.getId());
+                feature.setEnabledFor(featureItem.getConditions().getClientFilters());
+                featureSet.addFeature(feature);
+            } catch (IOException e) {
+                LOGGER.error("Unabled to parse Feature Management values from Azure.", e);
```

Comment:
There is a `failFast` property [here](https://github.com/microsoft/spring-cloud-azure/blob/master/spring-cloud-azure-appconfiguration-config/src/main/java/com/microsoft/azure/spring/cloud/config/AzureCloudConfigProperties.java#L57), thinking will it be better we throw the exception if is `failFast`, if not `failFast`, log exception only?

Reply:
Updated, see lines 87-89.